src/js/angular/core/directives/autocomplete/autocomplete.directive.js

```diff
@@ -168,47 +168,53 @@ function autocomplete($location, toastr, ClassInstanceDetailsService, Autocomple
+         * the function exits early to avoid repositioning elements unnecessarily.
          *
-         * If there's enough space in the viewport, it aligns the dropdown left edge
-         * with the left edge of the input field. This ensures that the dropdown is fully
-         * visible within the current viewport.
+         * If repositioning is needed, the function calculates the position of the `.autocomplete-results-wrapper`
+         * element by comparing the right edges of the dropdown and the viewport. If the dropdown extends
+         * beyond the viewport, it adjusts the CSS to align the dropdown's right edge with the right edge
+         * of the `edit-rule-row` element, taking into account the relative position of the `autocomplete-select`
+         * input. This adjustment ensures that the dropdown does not extend outside the viewport.
          *
-         * The function is wrapped inside a `$timeout` to ensure that the DOM has finished
-         * updating before calculating the positions and applying any adjustments.
+         * If there's sufficient space in the viewport, the dropdown's left edge is aligned with the left edge
+         * of its container. This logic ensures the dropdown is fully visible within the current viewport,
+         * enhancing the user interface experience.
          *
-         * Additionally, an angular watch is set up on `autoCompleteUriResults` to reposition
-         * the dropdown or reset its position based on the presence or absence of autocomplete results.
+         * This function is executed within a `$timeout` to ensure that any DOM updates are completed
+         * before calculating positions and applying adjustments. Additionally, an angular watch
+         * on `autoCompleteUriResults` triggers this function to either reposition the dropdown or
+         * reset its position, depending on the presence or absence of autocomplete results.
          *
-         * @param {boolean} reset - if we should reset position;
+         * @param {boolean} reset - Indicates whether to reset the dropdown's position.
          */
         $scope.positionAutocompleteWrapper = function(reset) {
             $timeout(() => {
-                const dropdown = element.find('.autocomplete-results-wrapper').get(0);
+                const dropdownElement = element.find('.autocomplete-results-wrapper');
+                const dropdown = dropdownElement.get(0);
                 if (reset) {
                     dropdown.style.left = 'auto';
                     dropdown.style.right = 'auto';
                     return;
                 }
-
-                if (dropdown && dropdown.style.right === '0px') {
+                if (dropdown && dropdown.style.right !== 'auto') {
                     return;
                 }
 
+                const autocompleteInputRect = dropdownElement.prevObject[0].getElementsByClassName('autocomplete-select')[0].getBoundingClientRect();
```

Comment:
This is some weird traversal. I'm not sure where this `prevObject` comes from? Was it something `jQuery` adds? Can't you just use plain DOM functions like `dropdownElement.previousSibling` to get the input field or `dropdownElement.parent.querySelector('.autocomplete-select')` if you are conserned that the input might not be always the previous sibling element